src/units.js

```diff
@@ -94,5 +94,6 @@ export const calculateSize = function(
             scale *= unitOptions.sizeMultiplier / options.sizeMultiplier;
         }
     }
-    return Math.min(sizeValue.number * scale, options.maxSize);
+    const scaled = Math.min(sizeValue.number * scale, options.maxSize);
+    return +scaled.toFixed(4);
```

Comment:
Maybe we can add the TeXbook reference in a comment to justify the choice of 4?

Reply:
Sounds good, added your observation and link in there.

Reply:
Instead of converting to fixed units here, I think we should consider doing the conversion in `toNode` and `toMarkup` in domTree.js.  This would mean we wouldn't have to make the changes we're making to mathMLTree.js.

Reply:
@kokopelli314 this would also address the issue you raised:
> It looks like there are some DOM tree elements that don't run their styles through calculateSize, meaning they're not rounded now. In particular I've noticed vlist and strut are not rounded. Would this require a more comprehensive refactor of the buildCommon.makeVList function?

Reply:
@kevinbarabash I think it's better to handle rounding when calculating the number. If we do it in `toNode`, we need to detect numbers in string, involving converting the string back to number and may give false positives.

Reply:
That's a good point.  I forgot that they're already strings by time we get to `toNode`.

Reply:
@kevinbarabash The PR was originally done that way, and it was messier; see https://github.com/Khan/KaTeX/pull/1747/files/bcd69c54ad850a8149c247ac5995b7e92c396591

It's a little sad that this approach (still) converts from number to string to number and later back to string, but I guess it'd be hard to return a string here...